internal/mesh/internal/controllers/sidecarproxy/builder/destinations.go

```diff
@@ -337,7 +337,7 @@ func (b *Builder) buildDestination(
 			}
 		}
 
-		b.addCluster(clusterName, endpointGroups, connectTimeout)
+		b.addCluster(clusterName, endpointGroups, connectTimeout, pbproxystate.Protocol(effectiveProtocol))
```

Comment:
pbproxystate.Protocol has the same enums as pbcatalog.Protocol so rather than having to maintain our own map of values we can directly call this.